UsbIpServer/ProcessUtils.cs

```diff
@@ -33,19 +71,45 @@ public static async Task<ProcessResult> RunCapturedProcessAsync(string filename,
             var stdout = string.Empty;
             var stderr = string.Empty;
 
-            await Task.WhenAll(
+            var captureTasks = new[]
+            {
                 Task.Run(async () => { stdout = await process.StandardOutput.ReadToEndAsync(); }, cancellationToken),
                 Task.Run(async () => { stderr = await process.StandardError.ReadToEndAsync(); }, cancellationToken),
-                process.WaitForExitAsync(cancellationToken));
+            };
+
+            try
+            {
+                await process.WaitForExitAsync(cancellationToken);
+            }
+            finally
+            {
+                await TerminateProcess(process);
+            }
 
+            // Since the local process either completed or was killed, these should complete or cancel promptly. 
+            foreach (var captureTask in captureTasks)
```

Comment:
This can still use `Task.WhenAll()`, although I don't think the ordering is important.

Reply:
You're right. WhenAll is better coding style here.